src/workload.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Implementation of WorkloadBase for running on VMs."""
+
+import logging
+from pathlib import Path
+
+from charms.operator_libs_linux.v2 import snap
+from typing_extensions import override
+
+from core.workload import WorkloadBase
+from literals import SNAP_NAME, SNAP_REVISION, SNAP_SERVICE
+
+logger = logging.getLogger(__name__)
+
+
+class EtcdWorkload(WorkloadBase):
+    """Implementation of WorkloadBase for running on VMs."""
+
+    def __init__(self):
+        self.etcd = snap.SnapCache()[SNAP_NAME]
```

Comment:
we should have a retry mechanism here, this is often prone to failing the first time - similar to [here](https://github.com/canonical/opensearch-operator/blob/2/edge/src/opensearch.py#L52)

Reply:
Done.